### PR TITLE
Lgat subset

### PR DESCRIPTION
**Pull request by kgaonkar6**
### Purpose/implementation Section
Subset metadata and consensus maf file to biospecimen where short_histology == LGAT

#### What scientific question is your analysis addressing?
Find biospecimens which are Low grade and found in consensus maf file

#### What was your approach?
I used short_histology == LGAT ,composition == "Solid Tissue",experimental_strategy == "WGS" and sample_type == "Tumor" to find biospecimens that are of interest here.
Then I check for BRAF V600E mutation and make a column BRAF_V600E which is Yes or No depending on if BRAF V600E mutation is present in the biospecimen.

#### What GitHub issue does your pull request address?
#631

#### Which areas should receive a particularly close look?
Sample selection and BRAF mutation checks

#### Is there anything that you want to discuss further?
Do we want the subset the consensus file to all mutations in LGAT samples or like I have it here subsetting samples and adding BRAF V600E specific information?

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
table in 
lgat-subset/LGAT_snv_subset.tsv

#### What types of results are included (e.g., table, figure)?
table

#### What is your summary of the results?
247 WGS lgat biospecimen with BRAF V600E status 

Not to be added to CI
#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# Subsets the consensus mutation file to biospecimens where short_histology == LGAT
+
+# Look for git root folder
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# get subset folder
+subset_dir<-file.path(root_dir,"analyses","molecular-subtyping-LGAT","lgat-subset")
+
+# clinical file
+clinical<- read_tsv(file.path(root_dir,"data","pbta-histologies.tsv"))
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R]: I would expect to see:

```suggestion
clinical<- readr::read_tsv(file.path(root_dir,"data","pbta-histologies.tsv"))
```

File: analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# consensus mutation data
+consensusMutation <-read_tsv(file.path(root_dir,"data","pbta-snv-consensus-mutation.maf.tsv.gz"))
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R]: Same comment
```suggestion
consensusMutation <- readr::read_tsv(file.path(root_dir,"data","pbta-snv-consensus-mutation.maf.tsv.gz"))
```

File: analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+consensusMutationSubset <- consensusMutation %>%
+  # find lgat samples
+  dplyr::filter(Tumor_Sample_Barcode %in% lgat_subset$Kids_First_Biospecimen_ID) %>%
+  # look for BRAF V600E mutations and make a column for BRAF_V600E
+  dplyr::filter(Hugo_Symbol == "BRAF" & HGVSp_Short == "p.V600E") %>%
+  # select tumor sample barcode
+  dplyr::select(Tumor_Sample_Barcode,HGVSp_Short) %>% 
+  unique() %>%
+  # join other WGS LGAT samples
+  full_join(lgat_wgs_subset,by=c("Tumor_Sample_Barcode"="Kids_First_Biospecimen_ID")) %>%
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R]: I don't know if you want to join all of the histology data here. I'd probably add a `select()` in here where you only grab the column you need – I'm under the impression that you need the biospecimen IDs in order to "fill out" the rest of the cases _without_ BRAF V600E and not much else at this stage, did I follow correctly?

File: analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# K S Gaonkar 2020
+# Subsets the consensus mutation file to biospecimens where short_histology == LGAT
+
+# Look for git root folder
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# get subset folder
+subset_dir<-file.path(root_dir,"analyses","molecular-subtyping-LGAT","lgat-subset")
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R]: Let's make sure this directory gets created if it doesn't exist with:
```r
if (!dir.exists(subset_dir)) {
  dir.create(subset_dir)
}
```

You're also able to do
```R
 dir.create(subset_dir, showWarnings = FALSE)
```
but I like how spelled out the other one is. A matter of personal preference.

File: analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  # join other WGS LGAT samples
+  full_join(lgat_wgs_subset,by=c("Tumor_Sample_Barcode"="Kids_First_Biospecimen_ID")) %>%
+  # get BRAF_V600E status
+  dplyr::mutate(BRAF_V600E=dplyr::case_when(HGVSp_Short=="p.V600E"~"Yes",is.na(HGVSp_Short) ~"No"))
+
+# remove consensusMutation
+rm(consensusMutation)
+
+# save to subset folder
+write_tsv(consensusMutationSubset,file.path(subset_dir, "LGAT_snv_subset.tsv"))
```
Comment (jaclyn-taroni) [inline analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R]:
```suggestion
readr::write_tsv(consensusMutationSubset,file.path(subset_dir, "LGAT_snv_subset.tsv"))
```

File: analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R (hunk 3 above)
Comment (kgaonkar6) [reply analyses/molecular-subtyping-LGAT/00-subset-files-for-LGAT.R]: yes that's right, I'll select only the biospecimen IDs here

**Pull request merged**